### PR TITLE
Add support for specifying unique hosted zone identifier

### PR DESCRIPTION
**Pull request by Pallokala**
## Add support for defining unique hosted zone identifier when modifying route53 hosted zone

With this change, the users get the possibility for defining with Hosted Zone ID which hosted zone they want to modify. This is important when there exists multiple identically named private hosted zones in different VPCs.

The change is (should be) fully backwards compatible.

### CODE REVIEW

**Comment by Pallokala** [comment]
Are there any news for this? It would really benefit me if this was merged soonish.

**Comment by gregdek** [comment]
@bpennypacker can you review this PR? If you give it +1 we will tag as "P1_reviewed" for quicker inclusion.

**Comment by bpennypacker** [comment]
+1 looks good!

**Comment by gregdek** [comment]
Thanks @bpennypacker

File: cloud/amazon/route53.py [hunk 1]
```diff
@@ -35,6 +35,12 @@
     required: true
     default: null
     aliases: []
+  hosted_zone_id:
```
Comment (bcoca) [inline cloud/amazon/route53.py:38]: needs version added and you can drop empty aliases

**Comment by gregdek** [comment]
@Pallokala back to you -- if you can add version (2.0) and drop empty aliases, we should be good to go.

**Comment by gregdek** [comment]
Oh, and apparently this also needs to be rebased now... quite a bit going on in this module...

**Comment by Pallokala** [comment]
I hopefully rebased this correctly and modified the patch to fit with the latest version.

**Pull request merged**